config/src/network_id.rs

```diff
@@ -2,14 +2,14 @@
 // SPDX-License-Identifier: Apache-2.0
 use crate::config::{PeerRole, RoleType};
 use diem_types::PeerId;
-use serde::{Deserialize, Serialize, Serializer};
+use serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};
 use short_hex_str::AsShortHexStr;
 use std::{cmp::Ordering, fmt, str::FromStr};
 
 /// A grouping of common information between all networking code for logging.
 /// This should greatly reduce the groupings between these given everywhere, and will allow
 /// for logging accordingly.
-#[derive(Clone, Eq, PartialEq, Serialize)]
+#[derive(Clone, Copy, Eq, PartialEq, Serialize)]
```

Comment:
oh man that's nice, we can throw out all those `Arc<NetworkContext>` clones everywhere :shipitparrot:

Reply:
I have a feeling `NetworkId` was a showstopper for a lot of copy vs clone problems